generate_route.py

```diff
@@ -46,30 +40,36 @@ class Location(object):
       longitude A float for the longitude of the location in Decimal Degrees
   """
 
-  def __init__(self,latitude,longitude):
+  def __init__(self,latitude,longitude,altitude=None):
     self.latitude = latitude
     self.longitude = longitude
-    self.altitude = None
+    self.altitude = altitude
     self.x = None
     self.y = None
     self.z = None
 
   def get_lat_lon_tuple(self):
     return (self.latitude,self.longitude)
 
-  def get_XYZ(Self):
-    pass
+  def get_xyz_tuple(self):
+    return (self.x,self.y,self.z)
+
+  def add_XYZ(self):
+    self.x,self.y,self.z = lla_to_xyz(self.latitude,self.longitude,self.altitude)
 
   def __repr__(self):
     return 'Location(%s, %s, %s)' % (self.latitude, self.longitude, self.altitude)
 
 class Route(object):
-  """An object for a route based on the input of a start and ending location
+  """An object for a route based on the input of a start and ending location.
 
   Attributes:
       start_location: a Location object for the start of the route
       end_location: a Location object for the end of the route
       route: a list of Location objects for each point on the route
+      distances: a list of distances between each pair of consecutive locations
+      durations: a list of durations between each pair of consecutive locations
```

Comment:
Both these are unused right now. Is there a plan to do something with them later?